src/main/java/org/truffleruby/parser/parser/ParserRopeOperations.java

```diff
@@ -61,6 +64,10 @@ public Rope makeShared(Rope rope, int sharedStart, int sharedLength) {
         }
     }
 
+    private static boolean isBuildingAOTImage() {
+        return RubyContext.FIRST_INSTANCE == null;
```

Comment:
Yeah. I am concerned that this test would be not robust and obscure without it.
Maybe it would be worth to have this as a static method and use it instead of various checks like context == null.